spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/extractors/TraceIdExtractor.java

```diff
@@ -32,4 +32,33 @@
 	 * @return The trace ID or null, if none found.
 	 */
 	String extractTraceIdFromRequest(HttpServletRequest req);
+
+	/**
+	 * Enum values represent the order and combination of predefined trace ID extractors that
+	 * can be used.
+	 *
+	 * @see XCloudTraceIdExtractor
+	 * @see ZipkinTraceIdExtractor
+	 * @see CompositeTraceIdExtractor
+	 */
+	enum Type {
```

Comment:
The things doesn't work this way. 
You are mixing here a concern of the `TraceIdExtractor` abstraction and possible out-of-the-box types.
The author of any possible custom implementation would be confused. 

If you would like to have it as an inner class, this `Type` should go into the `StackdriverLoggingProperties`.
Outside of the auto-configuration the `enum` does not make sense: you are not going to instantiate the target instance based on the `enum` manually.